<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1">



<title>PIME - Prevalence Interval for Microbiome Evaluation</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<link href="data:text/css;charset=utf-8,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23header%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%20code%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" rel="stylesheet" type="text/css" />




</head>

<body>




<h1 class="title toc-ignore">PIME - Prevalence Interval for Microbiome Evaluation</h1>



<div id="pime-prevalence-interval-for-microbiome-evaluation" class="section level1">
<h1>PIME: Prevalence Interval for Microbiome Evaluation</h1>
<p>PIME removes the within group variation found in metataxonomic surveys (16S rRNA datasets) by capturing only biological differences at high samples prevalence levels.</p>
</div>
<div id="first-steps-installing-phyloseq-and-creating-a-phyloseq-object" class="section level1">
<h1>First steps: installing ‘phyloseq’ and creating a ‘phyloseq’ object</h1>
<p>To install phyloseq start R and enter:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="cf">if</span> (<span class="op">!</span><span class="kw">requireNamespace</span>(<span class="st">&quot;BiocManager&quot;</span>, <span class="dt">quietly =</span> <span class="ot">TRUE</span>))
    <span class="kw">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)
BiocManager<span class="op">::</span><span class="kw">install</span>(<span class="st">&quot;phyloseq&quot;</span>)</code></pre></div>
<p>Phyloseq aggregates, mainly, an otu_table (with OTUs abundances), tax_table (taxonomy), and a sample_data (sample information) objects.</p>
<p>Importing output from DADA2</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">seqtab =<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">&quot;path_to_file/sequence_table_final.rds&quot;</span>)
tax=<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">&quot;path_to_file/tax_final.rds&quot;</span>)
map &lt;-<span class="st"> &quot;path_to_file/sample_data.txt&quot;</span>
ps &lt;-<span class="st"> </span><span class="kw">phyloseq</span>(<span class="kw">otu_table</span>(seqtab, <span class="dt">taxa_are_rows=</span><span class="ot">FALSE</span>), 
               <span class="kw">tax_table</span>(taxa))
sample_metadata =<span class="st"> </span><span class="kw">import_qiime_sample_data</span>(map)
physeq =<span class="kw">merge_phyloseq</span>(ps, sample_metadata)</code></pre></div>
<p>Importing otu_table as a biom file with taxonomy</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">jsonbiomfile =<span class="st"> &quot;path_to_file/otu_table_fix.biom&quot;</span>
mapfile =<span class="st"> &quot;path_to_file/v35_map_uniquebyPSN.txt&quot;</span>
biom =<span class="st"> </span><span class="kw">import_biom</span>(jsonbiomfile, mapfile, <span class="dt">parseFunction=</span>parse_taxonomy_greengenes)
map =<span class="st"> </span><span class="kw">import_qiime_sample_data</span>(mapfile)
input =<span class="st"> </span><span class="kw">merge_phyloseq</span>(biom,map)</code></pre></div>
<p>For further data input methods and examples go to <a href="https://joey711.github.io/phyloseq/import-data.html" class="uri">https://joey711.github.io/phyloseq/import-data.html</a></p>
</div>
<div id="how-to-install-pime-package" class="section level1">
<h1>How to install PIME package</h1>
<p>To install PIME first install the devtools package. Then load the library(devtools) and run install_github using the following commands.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#install.packages(&quot;devtools&quot;)</span>
<span class="kw">library</span>(devtools)
<span class="kw">install_github</span>(<span class="st">&quot;microEcology/pime&quot;</span>)</code></pre></div>
<p>PIME uses a Phyloseq object as input. A description of the phyloseq object and a tutorial on how to create this file in R using OTU tables in many different formats is detailed into the Phyloseq website <a href="https://joey711.github.io/phyloseq/" class="uri">https://joey711.github.io/phyloseq/</a></p>
</div>
<div id="step-by-step-example" class="section level1">
<h1>Step-by-step example</h1>
<p>The first step in PIME is to define if the microbial community presents a high relative abundance of taxa with low prevalence, which is considered as noise in PIME analysis. This is calculated by random forests analysis. In this example we run PIME using the restroom dataset (<a href="https://doi.org/10.1007%2Fs10482-017-0976-6">https://doi.org/10.1007%2Fs10482-017-0976-6</a>) against the metadata variable called Environment (a variable with two categories: men’s and women’s restroom).</p>
<p>Prediction using random forests on full dataset. Results in Out of Bag error rate. The input file was rarefied at 500 sequences for the purpose of this example (speed up the analysis). Using a rarefied dataset is recommended at this step.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(pime)
<span class="kw">data</span>(<span class="st">&quot;restroom&quot;</span>)
<span class="kw">pime.oob.error</span>(restroom, <span class="st">&quot;Environment&quot;</span>)
<span class="co">#&gt; [1] 0.6111111</span></code></pre></div>
<p>The OOB error rate &lt;=0.1, indicated the dataset present large differences, and pime might not remove much of the noise. Higher OOB error rate indicates that the next functions should be run to find the best prevalence interval for the dataset.</p>
<p>This function takes two parameters: The phyloseq object (restroom) and the predictor variable (Environment).</p>
<div id="split-the-dataset-by-predictor-variable" class="section level2">
<h2>Split the dataset by predictor variable</h2>
<p>Two parameters are required to run this function: The phyloseq object (restroom) and the predictor variable (Environment).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">per_variable_obj=<span class="st"> </span><span class="kw">pime.split.by.variable</span>(restroom, <span class="st">&quot;Environment&quot;</span>)
per_variable_obj
<span class="co">#&gt; $Restroom_F</span>
<span class="co">#&gt; phyloseq-class experiment-level object</span>
<span class="co">#&gt; otu_table()   OTU Table:         [ 1858 taxa and 9 samples ]</span>
<span class="co">#&gt; sample_data() Sample Data:       [ 9 samples by 6 sample variables ]</span>
<span class="co">#&gt; tax_table()   Taxonomy Table:    [ 1858 taxa by 7 taxonomic ranks ]</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Restroom_M</span>
<span class="co">#&gt; phyloseq-class experiment-level object</span>
<span class="co">#&gt; otu_table()   OTU Table:         [ 1983 taxa and 9 samples ]</span>
<span class="co">#&gt; sample_data() Sample Data:       [ 9 samples by 6 sample variables ]</span>
<span class="co">#&gt; tax_table()   Taxonomy Table:    [ 1983 taxa by 7 taxonomic ranks ]</span></code></pre></div>
</div>
<div id="calculate-the-highest-possible-prevalence-intervals" class="section level2">
<h2>Calculate the highest possible prevalence intervals</h2>
<p>This function calculates prevalence for different intervals by increments of 5. The input file is the output from the pime.split.by.variable(per_variable_obj)</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">prevalences=<span class="kw">pime.prevalence</span>(per_variable_obj)
<span class="kw">head</span>(prevalences)
<span class="co">#&gt; $`5`</span>
<span class="co">#&gt; phyloseq-class experiment-level object</span>
<span class="co">#&gt; otu_table()   OTU Table:         [ 3253 taxa and 18 samples ]</span>
<span class="co">#&gt; sample_data() Sample Data:       [ 18 samples by 6 sample variables ]</span>
<span class="co">#&gt; tax_table()   Taxonomy Table:    [ 3253 taxa by 7 taxonomic ranks ]</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`10`</span>
<span class="co">#&gt; phyloseq-class experiment-level object</span>
<span class="co">#&gt; otu_table()   OTU Table:         [ 3253 taxa and 18 samples ]</span>
<span class="co">#&gt; sample_data() Sample Data:       [ 18 samples by 6 sample variables ]</span>
<span class="co">#&gt; tax_table()   Taxonomy Table:    [ 3253 taxa by 7 taxonomic ranks ]</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`15`</span>
<span class="co">#&gt; phyloseq-class experiment-level object</span>
<span class="co">#&gt; otu_table()   OTU Table:         [ 593 taxa and 18 samples ]</span>
<span class="co">#&gt; sample_data() Sample Data:       [ 18 samples by 6 sample variables ]</span>
<span class="co">#&gt; tax_table()   Taxonomy Table:    [ 593 taxa by 7 taxonomic ranks ]</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`20`</span>
<span class="co">#&gt; phyloseq-class experiment-level object</span>
<span class="co">#&gt; otu_table()   OTU Table:         [ 593 taxa and 18 samples ]</span>
<span class="co">#&gt; sample_data() Sample Data:       [ 18 samples by 6 sample variables ]</span>
<span class="co">#&gt; tax_table()   Taxonomy Table:    [ 593 taxa by 7 taxonomic ranks ]</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`25`</span>
<span class="co">#&gt; phyloseq-class experiment-level object</span>
<span class="co">#&gt; otu_table()   OTU Table:         [ 222 taxa and 18 samples ]</span>
<span class="co">#&gt; sample_data() Sample Data:       [ 18 samples by 6 sample variables ]</span>
<span class="co">#&gt; tax_table()   Taxonomy Table:    [ 222 taxa by 7 taxonomic ranks ]</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`30`</span>
<span class="co">#&gt; phyloseq-class experiment-level object</span>
<span class="co">#&gt; otu_table()   OTU Table:         [ 222 taxa and 18 samples ]</span>
<span class="co">#&gt; sample_data() Sample Data:       [ 18 samples by 6 sample variables ]</span>
<span class="co">#&gt; tax_table()   Taxonomy Table:    [ 222 taxa by 7 taxonomic ranks ]</span></code></pre></div>
</div>
<div id="calculate-the-best-prevalence-interval-for-the-dataset" class="section level2">
<h2>Calculate the best prevalence interval for the dataset</h2>
<p>This function will return a table with Out of Bag error from random forests for each prevalence interval. The number of taxa and the number of remaining sequences for each prevalence interval are also computed. The best prevalence interval value provides the clearest separation of communities while still including a majority of the taxa in the analysis. If true differences are present. It will be represented by the first interval in which the OOB error rate is zero or close to zero. The input file is the list of prevalences generated by the pime.prevalence (prevalences) and the predictor variable (“Environment”“).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">set.seed</span>(<span class="dv">42</span>)
best.prev=<span class="kw">pime.best.prevalence</span>(prevalences, <span class="st">&quot;Environment&quot;</span>)
<span class="co">#&gt;        Interval OOB error rate (%) OTUs Nseqs</span>
<span class="co">#&gt;   Prevalence 5%              55.56 3253  9000</span>
<span class="co">#&gt;  Prevalence 10%                 50 3253  9000</span>
<span class="co">#&gt;  Prevalence 15%              11.11  593  5438</span>
<span class="co">#&gt;  Prevalence 20%              11.11  593  5438</span>
<span class="co">#&gt;  Prevalence 25%              11.11  222  4370</span>
<span class="co">#&gt;  Prevalence 30%              11.11  222  4370</span>
<span class="co">#&gt;  Prevalence 35%               5.56  117  3835</span>
<span class="co">#&gt;  Prevalence 40%               5.56  117  3835</span>
<span class="co">#&gt;  Prevalence 45%                  0   77  3531</span>
<span class="co">#&gt;  Prevalence 50%                  0   77  3531</span>
<span class="co">#&gt;  Prevalence 55%               5.56   77  3531</span>
<span class="co">#&gt;  Prevalence 60%                  0   45  3088</span>
<span class="co">#&gt;  Prevalence 65%                  0   45  3088</span>
<span class="co">#&gt;  Prevalence 70%                  0   26  2617</span>
<span class="co">#&gt;  Prevalence 75%                  0   26  2617</span>
<span class="co">#&gt;  Prevalence 80%                  0   16  2084</span>
<span class="co">#&gt;  Prevalence 85%                  0   16  2084</span>
<span class="co">#&gt;  Prevalence 90%                  0    4  1489</span>
<span class="co">#&gt;  Prevalence 95%                  0    4  1489</span></code></pre></div>
<p>In addition, it also returns the results from the random forests classification for each prevalence level. It includes SequenceID (OTU/ASV), Mean Decrease Accurracy (MDA) for each sample group, that is how much that SequenceID was important for classification of that group. The Mean Decrease Impurity (Gini Importance) and taxonomy are also included.</p>
<p>To get the table with OTU/ASV importance of the chosen prevalence interval. Pime keeps only the top 30 OTUs/ASVs, whith highest MDA.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">imp65=best.prev<span class="op">$</span><span class="st">`</span><span class="dt">Importance</span><span class="st">`</span><span class="op">$</span><span class="st">`</span><span class="dt">Prevalence 65</span><span class="st">`</span>
<span class="kw">head</span>(knitr<span class="op">::</span><span class="kw">kable</span>(imp65, <span class="dt">format=</span><span class="st">&quot;markdown&quot;</span>))
<span class="co">#&gt; [1] &quot;|SequenceID  | Restroom_F| Restroom_M| MeanDecreaseAccuracy| MeanDecreaseGini|Rank1       |Rank2             |Rank3                  |Rank4                |Rank5                  |Rank6               |Rank7    |&quot;</span>
<span class="co">#&gt; [2] &quot;|:-----------|----------:|----------:|--------------------:|----------------:|:-----------|:-----------------|:----------------------|:--------------------|:----------------------|:-------------------|:--------|&quot;</span>
<span class="co">#&gt; [3] &quot;|denovo87919 |  0.0909333|  0.0883333|            0.0804214|        1.3798848|k__Bacteria |p__Firmicutes     |c__Bacilli             |o__Lactobacillales   |f__Lactobacillaceae    |g__Lactobacillus    |s__iners |&quot;</span>
<span class="co">#&gt; [4] &quot;|denovo22521 |  0.0420333|  0.0257333|            0.0312325|        0.8396941|k__Bacteria |p__Proteobacteria |c__Gammaproteobacteria |o__Pseudomonadales   |f__Pseudomonadaceae    |g__Pseudomonas      |s__      |&quot;</span>
<span class="co">#&gt; [5] &quot;|denovo65044 |  0.0282000|  0.0372857|            0.0292992|        0.6619939|Unassigned  |NA                |NA                     |NA                   |NA                     |NA                  |NA       |&quot;</span>
<span class="co">#&gt; [6] &quot;|denovo6450  |  0.0302667|  0.0171000|            0.0215468|        0.4021410|k__Bacteria |p__Proteobacteria |c__Alphaproteobacteria |o__Sphingomonadales  |f__Sphingomonadaceae   |g__Sphingomonas     |s__      |&quot;</span>

<span class="co">#To get the table with OOB error results.</span>
<span class="co">#best.prev$`OOB error`</span></code></pre></div>
</div>
<div id="within-this-dataset-the-best-prevalence-interval-was-65" class="section level2">
<h2>Within this dataset the best prevalence interval was 65%</h2>
<p>To obtain the phyloseq object at this cutoff use the following command.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">prevalence.<span class="dv">65</span> =<span class="st"> </span>prevalences<span class="op">$</span><span class="st">`</span><span class="dt">65</span><span class="st">`</span>
prevalence.<span class="dv">65</span>
<span class="co">#&gt; phyloseq-class experiment-level object</span>
<span class="co">#&gt; otu_table()   OTU Table:         [ 45 taxa and 18 samples ]</span>
<span class="co">#&gt; sample_data() Sample Data:       [ 18 samples by 6 sample variables ]</span>
<span class="co">#&gt; tax_table()   Taxonomy Table:    [ 45 taxa by 7 taxonomic ranks ]</span></code></pre></div>
</div>
<div id="estimating-prediction-error" class="section level2">
<h2>Estimating prediction error</h2>
<p>To estimate error in prediction, we will use pime.error.prediction() to randomly assign treatments to samples and run random forests classification on each prevalence interval. The function returns a boxplot and a table with results of each classification error. For the purposes of this example we are running only 10 randomizations for saving time but we recommend at least 100 randomizations to obtain reliable results.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">randomized=<span class="kw">pime.error.prediction</span>(restroom, <span class="st">&quot;Environment&quot;</span>, <span class="dt">bootstrap =</span> <span class="dv">10</span>, <span class="dt">parallel =</span> <span class="ot">TRUE</span>, <span class="dt">max.prev =</span> <span class="dv">95</span>)
randomized<span class="op">$</span>Plot
randomized<span class="op">$</span><span class="st">'Table results'</span></code></pre></div>
<p>It is also possible to estimate the variation of OOB error with each prevalence interval filtering. This is done by running the random forests classification for n times, determined by the user. This function will return a boxplot figure and a table for each classification error.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">replicated.oob.error=<span class="st"> </span><span class="kw">pime.oob.replicate</span>(prevalences, <span class="st">&quot;Environment&quot;</span>, <span class="dt">bootstrap =</span> <span class="dv">10</span>, <span class="dt">parallel =</span> <span class="ot">TRUE</span>)</code></pre></div>
</div>
</div>
<div id="getting-help" class="section level1">
<h1>Getting Help</h1>
<p>Please contact us if you need any help: <em><a href="mailto:contact@brmicrobiome.org">contact@brmicrobiome.org</a></em></p>
</div>
<div id="pime-team" class="section level1">
<h1>PIME Team</h1>
<p>Luiz F. W. Roesch (Universidade Federal do Pampa - Brazil)</p>
<p>Priscila T. Dobbler (Universidade Federal do Pampa - Brazil)</p>
<p>Victor S. Pylro (Universidade Federal de Lavras - Brazil)</p>
<p>Bryan Kolaczkowski (University of Florida - United States of America)</p>
<p>Jennifer C. Drew (University of Florida - United States of America)</p>
<p>Eric W. Triplett (University of Florida - United States of America)</p>
</div>
<div id="citation" class="section level1">
<h1>Citation</h1>
<p>Roesch et al. (2018), PIME: including the concept of prevalence for uncovering differences in microbiome noised data. Frontiers, submitted.</p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
